### ISSUE TITLE
Server gets slow

### ISSUE THREAD

**Issue by Sasuke-Uchiha** [maddinat0r/samp-discord-connector#31]
After few times when changemode is used to change game mode the server get's very slow, Something that would need a complete restart. Nothing works even chat goes like 1 min in line etc.

Using latest version, attached the log too. 
[discord-connector.log](https://github.com/maddinat0r/samp-discord-connector/files/1338090/discord-connector.log)

**Comment by criticalerror99** [comment]
I have similar problem. After few minutes, messages on Discord not come and server works very slow. Restarting the server by killing and starting it resolves problem.

**Comment by maddinat0r** [comment]
@Sasuke-Uchiha I've looked through your log file, and everything seems alright. Did you send over your full log? Are you sure this plugin causes your slowdown?
@criticalerror99 Could you also please reproduce this and share your log here?

**Comment by criticalerror99** [comment]
@maddinat0r 
[discord-connector.log](https://github.com/maddinat0r/samp-discord-connector/files/1367158/discord-connector.log)
When the log returning errors, server have mega lag.

**Comment by maddinat0r** [comment]
@criticalerror99 Can you please try out v0.2.1? You're suffering from the same reconnect problem. The plugin is trying to reconnect very often, and this might lead to lag. Can you also tell me the specs of the hardware your server is running on?

**Comment by criticalerror99** [comment]
With v0.2.1 server is running few hours and problem semes to be solved. 0 errors in log.

My VPS:
VPS SSD 1
KVM
1 vCore Intel® Xeon® E5-2630 v3 2,4 GHz
2 GB RAM
SSD 10 GB
Local Raid 10

**Comment by maddinat0r** [comment]
@criticalerror99 Can you please share your current log file here? I'd like to compare it to the one you uploaded before it started working again.

**Comment by criticalerror99** [comment]
@maddinat0r 
[discord-connector.log](https://github.com/maddinat0r/samp-discord-connector/files/1369496/discord-connector.log)

**Comment by Sasuke-Uchiha** [comment]
Yes, for sure it's because of it. We tested it on another server Mini Games and it faced same issue, We only used to call it on FilterScriptInit as the server restarts every 4-5 mins for changing mission.  And discord is loaded as a filterscript.  But after you changed the function I think it gets called each time when server starts even with changemode or gmx

**Comment by Sasuke-Uchiha** [comment]
I tried with a server on my localhost, windows it worked fine. Then I went on trying with the same server on Linux.

The server was a normal one without gmx in it. It worked longer this time before sever get's very very slow, not stucked. As it took 3 mins for the server to register me joining the server/typing rcon exit and then leaving it. 

Finally, the logs show something. 
[errors.log](https://github.com/maddinat0r/samp-discord-connector/files/1754427/errors.log)
[warnings.log](https://github.com/maddinat0r/samp-discord-connector/files/1754428/warnings.log)
[discord-connector.log](https://github.com/maddinat0r/samp-discord-connector/files/1754429/discord-connector.log)

**Comment by maddinat0r** [comment]
I suspect debug logging being the massive slowdown here. I've disabled it in v0.2.2, could you try that version out and report back @Sasuke-Uchiha?

**Comment by Sasuke-Uchiha** [comment]
After update;
Errors.log
```
[03/05/18 06:49:38] [plugins/discord-connector] DCC_GetUserName: invalid user id '0' (discord.pwn:54)
[03/05/18 06:49:46] [plugins/discord-connector] DCC_GetUserName: invalid user id '0' (discord.pwn:54)
```
So nothing works from Discord to in game for now, i'll check from server to discord lag as it takes few hours before that happening. 

https://github.com/maddinat0r/samp-discord-connector/issues/37

**Comment by Sasuke-Uchiha** [comment]
Yes, it's perfect now. Just the issue stated above is left. Moreover, just a question does discord send messages parallel to the server? Or it slows down. It would be great if something like mysql_pquery could be done with discord that won't affect servers speed at all.

**Comment by maddinat0r** [comment]
I'll release a fix for this new issue this week.
The HTTP requests happen in a single network thread. So it's more like `mysql_tquery`. Sending them in parallel wouldn't increase performance by a lot, but rather create a lot of problems, mainly because of the rate limitation system Discord sets in place to prevent abuse.

**Comment by Yousha** [comment]
" Invalid user Id 0 "

Happens when users speak in channel X while game server just connected to channel Y
ex:
`gYChannel = DCC_FindChannelById(32453564654654343);`

and in `DCC_OnChannelMessage()` callback, plugin detects SOME messages from channel X
which doesnt exist in gamemode

@maddinat0r

**Comment by maddinat0r** [comment]
@Yousha Please create a new issue, as I don’t see any connection between your problem and this issue here.
